### PR TITLE
Docs: Fix editor component regex pattern.

### PR DESCRIPTION
**Pull request by tech4him1**
**- Summary**

The regex pattern in the example editor component was not actually a regex, so it wasn't working properly.

**- Test plan**
N/A: Docs.

**- Description for the changelog**
Docs: Fix editor component regex pattern.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: docs/extending.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ CMS.registerEditorComponent({
   // Fields the user need to fill out when adding an instance of the component
   fields: [{name: 'id', label: 'Youtube Video ID', widget: 'string'}],
   // Pattern to identify a block as being an instance of this component
-  pattern: youtube (\S+)\s,
+  pattern: /^\s?youtube (\S+)\s?/,
```
Comment (erquhart) [inline docs/extending.md]: I think using the Hugo style delimiters (`{{<`/`>}}`) makes sense here for example purposes - any reason to exclude them?

**Comment by erquhart** [comment]
@tech4him1 still adding delims?

**Comment by tech4him1** [comment]
@erquhart Sorry, got sidetracked. How's this?

**Comment by erquhart** [comment]
@tech4him1 note: had to revert due to https://github.com/netlify/netlify-cms/issues/749.

**Pull request merged**